python/TestHarness/CSVDiffer.py

```diff
@@ -10,9 +10,12 @@
 import os, re, math
 
 class CSVDiffer:
-    def __init__(self, test_dir, out_files, abs_zero=1e-11, relative_error=5.5e-6, gold_dir='gold'):
+    def __init__(self, custom_columns, custom_rel_err, custom_abs_zero, test_dir, out_files, abs_zero=1e-11, relative_error=5.5e-6, gold_dir='gold'):
```

Comment:
Can we move these new parameters to the end with a suitable default so existing calls don't break (and nicer calling semantics)? Ie something like:
`def __init__(self, test_dir, out_files, abs_zero=1e-11, relative_error=5.5e-6, gold_dir='gold', custom_columns=[], custom_rel_err=[], custom_abs_zero=[]):`

Maybe put some comments in on how these are to be used...